src/pathfinding_service/api.py

```diff
@@ -253,6 +271,54 @@ def get(
         return {"last_iou": None}, 404
 
 
+@add_schema
+@dataclass
+class FeedbackRequest:
+    """A HTTP request to FeedbackResource"""
+
+    token: UUID = field(metadata={"required": True})
+    status: str = field(
+        metadata={
+            "marshmallow_field": fields.Str(
+                validate=validate.OneOf(("success", "failure")), required=True
+            )
+        }
+    )
+    path: List[Address] = field(
+        metadata={"marshmallow_field": fields.List(ChecksumAddress, many=True), "required": True}
+    )
+    Schema: ClassVar[Type[marshmallow.Schema]]
+
+
+class FeedbackResource(PathfinderResource):
+    def post(
+        self, token_network_address: str  # pylint: disable=unused-argument
+    ) -> Tuple[dict, int]:
+        token_network = self._validate_token_network_argument(token_network_address)
+        feedback_request = self._parse_post(FeedbackRequest)
+        feedback_token = self.pathfinding_service.database.get_feedback_token(
+            token_id=feedback_request.token, token_network_address=token_network.address
+        )
+
+        # The client doesn't need to know whether the feedback was accepted or not,
+        # so in case the token is invalid we still return HTTP 200
+        if not feedback_token:
+            return {}, 200
+        if not feedback_token.is_valid():
```

Comment:
Since both of these will do the same thing, you could fold them into a single if `if not feedback_token or not feedback_token.is_valid()`